src/Symfony/Component/HttpFoundation/RequestMatcher/RequestMatcher.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\RequestMatcher;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * RequestMatcher compares a pre-defined set of checks against a Request instance.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class RequestMatcher extends ChainRequestMatcher
+{
+    private $host;
+    private $methods;
+    private $ips;
+    private $attributes;
+    private $schemes;
+
+    /**
+     * @param string|null $path
+     * @param string|null $host
+     * @param string[]    $methods
+     * @param string[]    $ips
+     * @param array       $attributes
+     * @param string[]    $schemes
+     */
+    public function __construct($path = null, $host = null, array $methods = array(), array $ips = array(), array $attributes = array(), array $schemes = array())
+    {
+        $matchers = array();
+        if (null !== $path) {
+            $matchers[] = new PathRequestMatcher($path, true);
```

Comment:
Not sure.. to me this class just factorizes a chain matcher (from scalars). If you have ready-to-use matchers, just chain them yourself :confused: 

And eventually the entire class may not be needed, if we built a chain in the security extension from config.

Same for the current `ExpressionRequestMatcher`, which extends the (current) generic request matcher and integrates the EL component additionally. Now we have 2 generic request matchers... :confused: 

With this PR i propse a _standalone_ expression request matcher, which you can combine with other matchers (using a chain).